sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -112,6 +113,22 @@ public SentryId captureEvent(SentryEvent event, @Nullable Scope scope, @Nullable
       return SentryId.EMPTY_ID;
     }
 
+    SentryEvent finalEvent = event;
+    scope.withSession(
+        s -> {
+          if (finalEvent.isCrashed()) {
+            if (s != null) {
+              s.addError();
+            }
+          }
+          if (finalEvent.getEnvironment() != null) {
```

Comment:
Don't think we can do this. Session data can't change other than `state` and `error count`.
So I guess we can only close the session here and hope that the environment and release was set during session start. Which should be the case since guideline is to restart session if either is changed.

Reply:
We don't want this to run if the event is cached (reading from cached events after app restart):

Move this stuff to line 86